.github/workflows/docs.yml

```diff
@@ -0,0 +1,31 @@
+name: Publish API documentation
+
+on:
+  workflow_dispatch:
+  push:
+    tags:
+      - v[0-9]+.[0-9]+.[0-9]+*
```

Comment:
Can we confirm tags are protected on this repo?

Reply:
We should also limit the secrets to an environment which is restricted to these tags.

Reply:
I don't have access to the exact rulesets without admin permissions but I am unable to delete tags via the GitHub UI so I'm guessing yes?

Reply:
Added the secrets to the `publish-docs` env, we should be good to go here 👍